File: javascript/src/com.spoonacular.client/com.spoonacular.client.model/GetMealPlanWeek200ResponseDaysInnerItemsInnerValue.js
/**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over 380,000 recipes, thousands of ingredients, 800,000 food products, and 100,000 menu items. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 1.1
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
 * The GetMealPlanWeek200ResponseDaysInnerItemsInnerValue model module.
 * @module com.spoonacular.client/com.spoonacular.client.model/GetMealPlanWeek200ResponseDaysInnerItemsInnerValue
 * @version 1.1
 */
class GetMealPlanWeek200ResponseDaysInnerItemsInnerValue {
    /**
     * Constructs a new <code>GetMealPlanWeek200ResponseDaysInnerItemsInnerValue</code>.
     * @alias module:com.spoonacular.client/com.spoonacular.client.model/GetMealPlanWeek200ResponseDaysInnerItemsInnerValue
     * @param servings {Number} 
     * @param id {Number} 
     * @param title {String} 
     * @param imageType {String} 
     */
    constructor(servings, id, title, imageType) { 
        
        GetMealPlanWeek200ResponseDaysInnerItemsInnerValue.initialize(this, servings, id, title, imageType);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj, servings, id, title, imageType) { 
        obj['servings'] = servings;
        obj['id'] = id;
        obj['title'] = title;
        obj['imageType'] = imageType;
    }

    /**
     * Constructs a <code>GetMealPlanWeek200ResponseDaysInnerItemsInnerValue</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:com.spoonacular.client/com.spoonacular.client.model/GetMealPlanWeek200ResponseDaysInnerItemsInnerValue} obj Optional instance to populate.
     * @return {module:com.spoonacular.client/com.spoonacular.client.model/GetMealPlanWeek200ResponseDaysInnerItemsInnerValue} The populated <code>GetMealPlanWeek200ResponseDaysInnerItemsInnerValue</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new GetMealPlanWeek200ResponseDaysInnerItemsInnerValue();

            if (data.hasOwnProperty('servings')) {
                obj['servings'] = ApiClient.convertToType(data['servings'], 'Number');
            }
            if (data.hasOwnProperty('id')) {
                obj['id'] = ApiClient.convertToType(data['id'], 'Number');
            }
            if (data.hasOwnProperty('title')) {
                obj['title'] = ApiClient.convertToType(data['title'], 'String');
            }
            if (data.hasOwnProperty('imageType')) {
                obj['imageType'] = ApiClient.convertToType(data['imageType'], 'String');
            }
        }
        return obj;
    }


}

/**
 * @member {Number} servings
 */
GetMealPlanWeek200ResponseDaysInnerItemsInnerValue.prototype['servings'] = undefined;

/**
 * @member {Number} id
 */
GetMealPlanWeek200ResponseDaysInnerItemsInnerValue.prototype['id'] = undefined;

/**
 * @member {String} title
 */
GetMealPlanWeek200ResponseDaysInnerItemsInnerValue.prototype['title'] = undefined;

/**
 * @member {String} imageType
 */
GetMealPlanWeek200ResponseDaysInnerItemsInnerValue.prototype['imageType'] = undefined;






export default GetMealPlanWeek200ResponseDaysInnerItemsInnerValue;

File: csharp/src/spoonacular/Model/GetMealPlanWeek200ResponseDaysInnerItemsInnerValue.cs
/*
 * spoonacular API
 *
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 1.1
 * Contact: mail@spoonacular.com
 * Generated by: https://github.com/openapitools/openapi-generator.git
 */


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = spoonacular.Client.OpenAPIDateConverter;

namespace spoonacular.Model
{
    /// <summary>
    /// GetMealPlanWeek200ResponseDaysInnerItemsInnerValue
    /// </summary>
    [DataContract(Name = "getMealPlanWeek_200_response_days_inner_items_inner_value")]
    public partial class GetMealPlanWeek200ResponseDaysInnerItemsInnerValue : IValidatableObject
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="GetMealPlanWeek200ResponseDaysInnerItemsInnerValue" /> class.
        /// </summary>
        [JsonConstructorAttribute]
        protected GetMealPlanWeek200ResponseDaysInnerItemsInnerValue() { }
        /// <summary>
        /// Initializes a new instance of the <see cref="GetMealPlanWeek200ResponseDaysInnerItemsInnerValue" /> class.
        /// </summary>
        /// <param name="servings">servings (required).</param>
        /// <param name="id">id (required).</param>
        /// <param name="title">title (required).</param>
        /// <param name="imageType">imageType (required).</param>
        public GetMealPlanWeek200ResponseDaysInnerItemsInnerValue(decimal servings = default(decimal), decimal id = default(decimal), string title = default(string), string imageType = default(string))
        {
            this.Servings = servings;
            this.Id = id;
            // to ensure "title" is required (not null)
            if (title == null)
            {
                throw new ArgumentNullException("title is a required property for GetMealPlanWeek200ResponseDaysInnerItemsInnerValue and cannot be null");
            }
            this.Title = title;
            // to ensure "imageType" is required (not null)
            if (imageType == null)
            {
                throw new ArgumentNullException("imageType is a required property for GetMealPlanWeek200ResponseDaysInnerItemsInnerValue and cannot be null");
            }
            this.ImageType = imageType;
        }

        /// <summary>
        /// Gets or Sets Servings
        /// </summary>
        [DataMember(Name = "servings", IsRequired = true, EmitDefaultValue = true)]
        public decimal Servings { get; set; }

        /// <summary>
        /// Gets or Sets Id
        /// </summary>
        [DataMember(Name = "id", IsRequired = true, EmitDefaultValue = true)]
        public decimal Id { get; set; }

        /// <summary>
        /// Gets or Sets Title
        /// </summary>
        [DataMember(Name = "title", IsRequired = true, EmitDefaultValue = true)]
        public string Title { get; set; }

        /// <summary>
        /// Gets or Sets ImageType
        /// </summary>
        [DataMember(Name = "imageType", IsRequired = true, EmitDefaultValue = true)]
        public string ImageType { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append("class GetMealPlanWeek200ResponseDaysInnerItemsInnerValue {\n");
            sb.Append("  Servings: ").Append(Servings).Append("\n");
            sb.Append("  Id: ").Append(Id).Append("\n");
            sb.Append("  Title: ").Append(Title).Append("\n");
            sb.Append("  ImageType: ").Append(ImageType).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return Newtonsoft.Json.JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            // Title (string) minLength
            if (this.Title != null && this.Title.Length < 1)
            {
                yield return new System.ComponentModel.DataAnnotations.ValidationResult("Invalid value for Title, length must be greater than 1.", new [] { "Title" });
            }

            yield break;
        }
    }

}
